staging/src/k8s.io/client-go/tools/cache/mutation_cache.go

```diff
@@ -148,11 +148,11 @@ func (c *mutationCache) ByIndex(name string, indexKey string) ([]interface{}, er
 		fn := c.indexer.GetIndexers()[name]
 		// Keys() is returned oldest to newest, so full traversal does not alter the LRU behavior
 		for _, key := range c.mutationCache.Keys() {
-			updated, ok := c.mutationCache.Get(key)
```

Comment:
doing the Get first is important to maintain the LRU ordering (see "Keys() is returned oldest to newest, so full traversal does not alter the LRU behavior" comment)